### PR TITLE
Exploit prevention metrics

### PR DESCRIPTION
**Pull request by uurien**
### What does this PR do?

Adds metrics and telemetry when rasp address is executed. It adds metrics in rootspan with duration and call counts, and send telemetry with eval counts, matches or timeouts.

### Motivation

Have information about the rasp behaviour.

### Plugin Checklist

- [x] Unit tests.

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts

### Additional Notes
System tests: https://github.com/DataDog/system-tests/pull/2534

[APPSEC-53372]

[APPSEC-53372]: https://datadoghq.atlassian.net/browse/APPSEC-53372?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/dd-trace/test/appsec/telemetry.spec.js [hunk 1]
```diff
@@ -315,6 +432,44 @@ describe('Appsec Telemetry metrics', () => {
...
+        expect(raspEvalCount).to.be.eq(2)
+      })
+
+      it('should not increment any metric if telemetry metrics are disabled', () => {
+        appsecTelemetry.enable({
+          enabled: true,
+          metrics: false
+        })
+
+        appsecTelemetry.updateWafRequestsMetricTags({
```
Comment (CarlesDD) [inline packages/dd-trace/test/appsec/telemetry.spec.js:465]:
```suggestion
        appsecTelemetry.updateRaspRequestsMetricTags({
```

Comment (CarlesDD) [inline packages/dd-trace/test/appsec/telemetry.spec.js:465]:
```suggestion
        appsecTelemetry.updateRaspRequestsMetricTags({
```

Comment (uurien) [reply packages/dd-trace/test/appsec/telemetry.spec.js:465]: yes, I'll modify it.
And I realise that I wrote the tests of `updateRaspRequestsMetricTag` under `    describe('updateWafRequestMetricTags', () => {`, and I should've created new describe section, I'll modify it.

File: packages/dd-trace/test/appsec/waf/index.spec.js [hunk 2]
```diff
@@ -94,6 +94,32 @@ describe('WAF Manager', () => {
     })
   })
 
+  describe('run', () => {
+    it('should call wafManager.run with params', () => {
```
Comment (simon-id) [inline packages/dd-trace/test/appsec/waf/index.spec.js:98]:
```suggestion
    it('should call wafManager.run with raspRuleType', () => {
```

File: packages/dd-trace/test/appsec/telemetry.spec.js [hunk 3]
```diff
@@ -238,6 +304,34 @@ describe('Appsec Telemetry metrics', () => {
         })
       })
 
+      it('should not modify waf.requests metric tags when rasp rule type is provided', () => {
```
[the remark this answers is not in the document]
Comment (uurien) [reply packages/dd-trace/test/appsec/telemetry.spec.js:307]: Could it be clearer if I modify line 326 to
```javascript
expect(count.lastCall).to.have.been.calledWithExactly('waf.requests', {
```

??

[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/test/appsec/telemetry.spec.js:307]: mmm, maybe adding something like:
```javascript
expect(count).to.have.not.been.calledWithExactly('waf.requests', { true, true, true})
```
?

[the remark this answers is not in the document]
Comment (uurien) [reply packages/dd-trace/test/appsec/telemetry.spec.js:307]: I added a small check in the last commit, to be sure that count it is not called with `'waf.requests'` before `incrementWafRequestsMetric`. 
I think that now it is clear, that we are testing that `incrementWafRequestsMetric` is not affected by `updateRaspRequestsMetricTags`

[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/test/appsec/telemetry.spec.js:307]: good enough

**Review by simon-id — APPROVED** [review]

**Review by CarlesDD — APPROVED** [review]

**Pull request merged**